project/profiles/capacity4more/modules/c4m/user/c4m_user/actions/c4m_user_contact_multiple.action.inc

```diff
@@ -0,0 +1,164 @@
+<?php
+
+/**
+ * @file
+ * Callback & forms related to the user contact multiple.
+ */
+
+/**
+ * A custom action to contact a multiple users with a mail action.
+ *
+ * @return array
+ *   Information about the action.
+ */
+function c4m_user_contact_multiple_action_info() {
+  return array(
+    'c4m_user_contact_multiple_action' => array(
+      'type' => 'user',
+      'label' => t('Contact'),
+      'configurable' => TRUE,
+      'triggers' => array('any'),
+    ),
+  );
+}
+
+/**
+ * Create 'Contact Multiple' form.
+ */
+function c4m_user_contact_multiple_action_form(array $context, $form_state) {
+  global $user;
+  $to_list = '';
+
+  // If 'Select all items on all pages' is selected.
+  if ($form_state['select_all_pages']) {
+    $group = c4m_og_current_group();
+    // Get ids of all subscribers of current group.
+    $ids = og_get_group_members_properties($group, array(), 'members', 'node');
+  }
+  else {
+    // Get selected items on current page.
+    $ids = $form_state['selection'];
+  }
+
+  // Create a names list of notified subscribers.
+  foreach ($ids as $account_id) {
+    // Make sure that id is valid, and GO/GA didn't select himself.
+    if ($account_id != 0 && $user->uid != $account_id) {
+      $account = user_load($account_id);
```

Comment:
@anvmn 
we should use `user_load_multiple()` which will hit the DB only once, instead of hitting in for each user.